src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -32,4 +34,22 @@
  */
 public interface Extent extends EntityUniverse, WeatherUniverse, BlockVolume, BiomeArea {
 
+    /**
+     * Get a representation of the block at the given position.
+     *
+     * @param position The position
+     * @return The block
+     */
+    BlockLoc getFullBlock(Vector3i position);
```

Comment:
BlockState contains most of the block information, but is location independent and doesn't do stuff like chest inventories. Full block returns a BlockLoc, which is a state with a location that encapsulates all the block information. It's the best nomenclature I could come up considering that I wanted BlockVolume to keep its method names as "getBlock".